File: wirtualny_swiat/src/swiat/rosliny/Guarana.java
package swiat.rosliny;

import swiat.Organizm;
import swiat.Swiat;

import java.awt.Color;

public class Guarana extends Roslina {
    public Guarana(final int polozenieX, final int polozenieY, final int wiek) {
        setNazwa("Guarana");
        setSymbol('G');
        setSila(0);
        setInicjatywa(0);
        setWiek(wiek);
        setPolozenieX(polozenieX);
        setPolozenieY(polozenieY);
        setNiesmiertelnosc(false);
    }


    public Guarana(final Organizm inny) {
       kopiujObiekt(inny);
    }


    @Override
    public Organizm stworzNowySklonowanyObiekt() {
        return new Guarana(this);
    }


    @Override
    public void kolizja(Swiat swiat, Organizm atakujacy) {
        int silaAtakujacego = atakujacy.getSila();
        int polozenieX = getPolozenieX(), polozenieY = getPolozenieY(), polozenieXAtak = atakujacy.getPolozenieX(), polozenieYAtak = atakujacy.getPolozenieY();

        setCzyZyje(false);
        swiat.usunOrganizm(this, polozenieX, polozenieY);
        atakujacy.setPolozenieX(polozenieX);
        atakujacy.setPolozenieY(polozenieY);
        atakujacy.setSila(silaAtakujacego + 3);
        swiat.dodajOrganizm(atakujacy, polozenieX, polozenieY);
        swiat.usunOrganizm(atakujacy, polozenieXAtak, polozenieYAtak);
        swiat.wypiszWiadomosc(atakujacy.getNazwa() + " zjada Guarana " + polozenieX + " " + polozenieY);
    }


    @Override
    public Color rysowanie() {
        return Color.pink;
    }
}

File: wirtualny_swiat_python/Guarana.py
from Roslina import Roslina


class Guarana(Roslina):
    def __init__(self, x, y, wiek):
        super().__init__(x, y, wiek)
        self.setNazwa("Guarana")
        self.setSymbol('G')
        self.setSila(0)
        self.setInicjatywa(0)
        self.setNiesmiertelnosc(False)

    def stworzNowySklonowanyObiekt(self):
        return Guarana(0, 0, 0)

    def kolizja(self, swiat, atakujacy):
        atakujacy.setSila(atakujacy.getSila() + 3)
        super().kolizja(swiat, atakujacy)

    def rysowanie(self):
        return "hotpink"
